keps/sig-node/3570-cpumanager/README.md

```diff
@@ -323,6 +407,236 @@ func (p *staticPolicy) RemoveContainer(s State, containerID string) error {
+
+NOTE: in order to enable the feature, the cluster admin needs also to enable
+the `static` cpu manager policy.
+
+###### Does enabling the feature change any default behavior?
+
+No, unless the non-none policy (`static`) is explicitly configured.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, using the kubelet config.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The impact is node-local only.
+If the state of a node is steady, no changes.
+If a guaranteed pod is admitted, running non-guaranteed pods will have their CPU cgroup changed while running.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, covered by e2e tests
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollout can fail if a bug in the cpumanager prevents _new_ pods to start, or existing pods to be restarted.
+Already running workload will not be affected if the node state is steady
+
+###### What specific metrics should inform a rollback?
+
+Pod creation errors on a node-by-node basis.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No to both.
+Changes in behavior only affects pods meeting the conditions (guaranteed QoS, integral CPU request) scheduler after the upgrade.
+Running pods will be unaffected by any change. This offers some degree of safety in both upgrade->rollback
+and upgrade->downgrade->upgrade scenarios.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+Monitor the pod admission counter
+Monitor the pods not going running after successful schedule
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The operator need to inspect the node and verify the cpu pinning assignment either checking the cgroups on the node
+or accessing the podresources API of the kubelet.
+
+###### How can someone using this feature know that it is working for their instance?
+
+
+- [X] Other (treat as last resort)
+  - Details: the containers need to check the cpu set they are allowed to run; in addition, node agents (e.g. node_exporter)
+    can report the CPU assignment
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+- N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [ ] Other (treat as last resort)
+  - Details:
+     a operator should check that pods go running correctly and the cpu pinning is performed. The latter can
+     be checked by inspecting the cgroups at node level.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+No, because all the metrics we were aware of leaked hardware details.
```

Comment:
Are there are Pod metrics around # of Guarenteed pods? It seems like a metric could be derived from checking if a Pod is Guaranteed and has an integer CPU request/limit?

Reply:
Oh, I see this was already brought up here https://github.com/kubernetes/enhancements/pull/3350#discussion_r985983166

Reply:
yes, I elaborated this point after further discussion, in the same direction you mentioned.